internal/pkg/cli/init.go

```diff
@@ -546,6 +561,50 @@ func (o *initOpts) askShouldDeploy() error {
 	return nil
 }
 
+func (o *initOpts) askEnvName() error {
```

Comment:
I feel like since we've been already asking "Would you like to deploy a test environment?", this additional "asking for env name" flow doesn't quite make sense to me. Maybe we just need to introduce a new env name flag? So that they can use other env name than "test" (we also inject the env name if exists in `initShouldDeployPrompt`).

Or the bigger picture is to reuse `initOpts.Execute` for `copilot deploy`?

Reply:
Personally I don't love that we just default to deploying to `test`, while also initializing that environment if it doesn't exist. I think it has the possibility of creating some confusion where a customer has a `dev` environment set up, then a new developer runs `copilot init --deploy` and ends up creating a whole new environment, perhaps assuming that they'd get a chance to customize where the service is deployed. 

I want init and deploy to be very similar, in the sense that `init --env dev --deploy` will initialize and deploy an environment (if necessary) + the service, and that `deploy --deploy-env dev --init` will initialize and deploy the environment (if necessary) + the service. 

But `deploy` without any flags will be a guided workflow to deploy an environment or service and `init` without any flags will be a guided workflow to initialize and maybe deploy.